### DISCUSSION TITLE
Restrict mrpack path format syntax

### DISCUSSION THREAD

**Discussion by nothub** [modrinth/.github#45] [Feature requests]
### Is your suggested enhancement related to a problem? Please describe.

Currently, there is no restriction for the path format of files in a mrpacks manifest.

In the docs there are examples for avoiding path traversal attacks: `[A-Z]:/`, `[A-Z]:\`, and `/`. From this I understand that the path format syntax is not restricted to a common type.

By not restricting the path format syntax, the interpretation of paths defined in mrpack files will slightly differ between tools because the developers can implement it on their own discretion.

In the long term, this will lead to a path format syntax that is restricted by the community, based on the paths interpretation of popular tooling.

### Describe the solution you'd like

I suggest restricting the allowed path syntax to solely unix path syntax.

### Additional context

Related to [#84](https://github.com/modrinth/docs/issues/84)

Conversation on Discord:
[15:44] Emma (triphora): It should probably only be /
[15:44] Emma (triphora): Could you make a docs issue?

**Comment by Akarys42** [comment]
That'd be a breaking change. I think this would warrant a `fileVersion` bump, and all the version 1files should just have their `\` replaced by a `/`.

**Comment by nothub** [comment]
An example of `\\` path separators in the wild: https://cdn.modrinth.com/data/xldzprsQ/versions/MqYT8IQ2/Create%20Extra%20Full%201.2.0.mrpack

It should be evaluated if `\\` should be converted to `/` or any `\` should be converted to `/`.
Because of the current spec, the file path could also contain characters escaped by a `\` prefix.
